diff --git a/src/object/object-set.cpp b/src/object/object-set.cpp
index c4417ed..585d333 100644
--- a/src/object/object-set.cpp
+++ b/src/object/object-set.cpp
@@ -255,13 +255,9 @@ Inkscape::XML::Node *ObjectSet::topRepr() const
         return nullptr;
     }
 
-#ifdef __APPLE__
     // workaround for
     // static_assert(__is_cpp17_forward_iterator<_ForwardIterator>::value
     auto const n = std::vector<Inkscape::XML::Node *>(nodes.begin(), nodes.end());
-#else
-    auto const& n = nodes;
-#endif
 
     return *std::max_element(n.begin(), n.end(), sp_repr_compare_position_bool);
 }
