diff --git a/efi/generate_binary.py b/efi/generate_binary.py
index a8a4437..0c29528 100755
--- a/efi/generate_binary.py
+++ b/efi/generate_binary.py
@@ -43,7 +43,8 @@ def _run_objcopy(args):
         # "efi-app-*"
         argv.extend(["--output-target", "efi-app-{}".format(args.arch)])
     else:
-        argv.extend(["--target", "efi-app-{}".format(args.arch)])
+        argv.extend(["--input-target", "efi-app-{}".format(args.arch)])
+        #argv.extend(["--target", "efi-app-{}".format(args.arch)])
 
     try:
         subprocess.run(argv, check=True)
