diff --git a/libcap/Makefile b/libcap/Makefile
index 5b0db8f..82092d7 100644
--- a/libcap/Makefile
+++ b/libcap/Makefile
@@ -20,7 +20,7 @@ PSXMAGICOBJ=psx_magic.o
 
 # The linker magic needed to build a dynamic library as independently
 # executable
-MAGIC=--entry=__so_start
+MAGIC=-Wl,--entry=__so_start
 
 INCLS=libcap.h cap_names.h $(INCS)
 GPERF_OUTPUT = _caps_output.gperf
diff --git a/pam_cap/Makefile b/pam_cap/Makefile
index a15feae..50e007c 100644
--- a/pam_cap/Makefile
+++ b/pam_cap/Makefile
@@ -24,7 +24,7 @@ execable.o: execable.c ../libcap/execable.h ../libcap/loader.txt
 	$(CC) $(CFLAGS) $(LDFLAGS) $(CPPFLAGS) -DLIBCAP_VERSION=\"libcap-$(VERSION).$(MINOR)\" -DSHARED_LOADER=\"$(shell cat ../libcap/loader.txt)\" -c execable.c -o $@
 
 pam_cap.so: pam_cap.o execable.o
-	$(LD) $(CFLAGS) -o pam_cap.so $+ $(LIBCAPLIB) $(LDFLAGS) --entry=__so_start
+	$(LD) $(CFLAGS) -o pam_cap.so $+ $(LIBCAPLIB) $(LDFLAGS) -Wl,--entry=__so_start
 
 pam_cap.o: pam_cap.c
 	$(CC) $(CFLAGS) $(CPPFLAGS) -c $< -o $@
