diff --git a/libcap/Makefile b/libcap/Makefile
index e2b7a90..5b0db8f 100644
--- a/libcap/Makefile
+++ b/libcap/Makefile
@@ -118,7 +118,7 @@ empty: empty.c
 	$(CC) -o $@ $<
 
 loader.txt: empty
-	$(OBJCOPY) --dump-section .interp=/dev/stdout $< > $@
+	$(OBJCOPY) --dump-section .interp=$@ $<
 
 cap_magic.o: execable.h execable.c loader.txt
 	$(CC) $(CFLAGS) $(IPATH) -DLIBRARY_VERSION=\"$(LIBTITLE)-$(VERSION).$(MINOR)\" -DSHARED_LOADER=\"$(shell cat loader.txt)\" -c execable.c -o $@
