diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b6ac26b..da1ce52 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -123,7 +123,7 @@ if  (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
         if (GRAPHITE2_SANITIZERS)
             target_link_libraries(graphite2 c gcc_s)
         else ()
-            target_link_libraries(graphite2 c gcc)
+            target_link_libraries(graphite2 c)
         endif ()
     endif()
     include(Graphite)
